### ISSUE TITLE
Reactive key unwrapping

### ISSUE THREAD

**Issue by ItMaga** [DamianOsipiuk/vue-query#150]
The `hashQueryKeyByOptions` function returns a `queryHash` without unwrapping, so the `queryKey` check becomes false.

```ts
const QUERY_KEY = 'test';
const page = ref(1);
const queryKey = reactive([QUERY_KEY, { page }]);

const query = useQuery(queryKey, fetch);
....

queryClient.refetchQueries(queryKey, { active: true, exact: true });
```

![image](https://user-images.githubusercontent.com/51162910/160082588-1399f865-c706-4e01-b501-5712e7da8874.png)

UPD: version ^1.20.0

**Comment by DamianOsipiuk** [comment]
This should not be the case as `useQuery` should already unwrap the `queryKey`.
Could you re-test it with 
```js
const queryKey = [QUERY_KEY, { page }];
```
?

Also minimal reproducible example would help with debugging

**Comment by ItMaga** [comment]
> I have run your example and it seems to work as expected. You might need to provide reproduction repository to narrow down the issue.

I may have missed an important detail. This is reproduced when linking Vue2 + CompositionAPI.

[Reproduction link](https://github.com/ItMaga/vue-query-bug)

**Comment by DamianOsipiuk** [comment]
Actually it was broken for both Vue 2 and 3. And it worked for me due to specific setup. Should be fixed in [v1.22.2](https://github.com/DamianOsipiuk/vue-query/releases/tag/v1.22.2)

**Comment by ItMaga** [comment]
It works, thanks @DamianOsipiuk)
